cpp/src/io/comp/cpu_unbz2.cpp

```diff
@@ -88,41 +88,53 @@ For more information on these sources, see the manual.
 namespace cudf {
 namespace io {
 #ifdef _MSC_VER
-#define bswap_32(v) _byteswap_ulong(v)
-#define bswap_64(v) _byteswap_uint64(v)
-#else
-#define bswap_32(v) __builtin_bswap32(v)
-#define bswap_64(v) __builtin_bswap64(v)
-#endif
 
-// Constants for the fast MTF decoder.
-#define MTFA_SIZE 4096
-#define MTFL_SIZE 16
+template <typename T>
+inline T bswap_32(T val)
+{
+  return _byteswap_ulong(val);
+}
 
-// Header bytes.
-#define BZ_HDR_B 0x42 /* 'B' */
-#define BZ_HDR_Z 0x5a /* 'Z' */
-#define BZ_HDR_h 0x68 /* 'h' */
-#define BZ_HDR_0 0x30 /* '0' */
+template <typename T>
+inline T bswap_64(T val)
+{
+  return _byteswap_uint64(val);
+}
 
-// Constants for the back end.
+#else
 
-#define BZ_MAX_ALPHA_SIZE 258
-#define BZ_MAX_CODE_LEN 23
+template <typename T>
+inline T bswap_32(T val)
+{
+  return __builtin_bswap32(val);
+}
+
+template <typename T>
+inline T bswap_64(T val)
+{
+  return __builtin_bswap64(val);
+}
```

Comment:
How about this instead. 
```suggestion
template <typename T>
inline std::enable_if_t< sizeof(T) = 4, T> byte_swap(T val)
{
  return __builtin_bswap32(val);
}

template <typename T>
inline std::enable_if_t<sizeof(T) = 8, T>  byte_swap(T val)
{
  return __builtin_bswap64(val);
}
```